sql/core/src/main/scala/org/apache/spark/sql/api/python/PythonSQLUtils.scala

```diff
@@ -126,6 +127,14 @@ private[sql] object PythonSQLUtils extends Logging {
     deserializer(internalRow)
   }
 
+  def addJarToCurrentClassLoader(path: String): Unit = {
```

Comment:
Got this logic from my own (unmerged) PR long time ago: https://github.com/apache/spark/pull/19643.
This code path will only be used in development, and cannot be used in production. I will clean up with some docs later.

Reply:
Question though:

Does this util help if we want to support the `application_jar` in spark-submit if we enable Connect there?

Reply:
Yeah we will have to support other options in the future. But this logic would have to stay here because we're automatically searching the jars (dev mode only)

Reply:
ack. This class is private to sql package so defining this function seems to be safe for internal use only (`private[sql] object PythonSQLUtils`)